test/unit/proxy_test.rb

```diff
@@ -159,6 +169,14 @@ def test_sandbox_endpoint_validation
       proxy.staging_endpoint = ''
 
       assert proxy.valid?
+
+      proxy.sandbox_endpoint = "http://localhost:8080"
+      assert proxy.valid?
+
+      service.deployment_option = 'hosted'
+      service.save
+      proxy.sandbox_endpoint = "http://localhost:8080"
+      refute proxy.valid?
```

Comment:
as far as I see we standardize on `assert_not`. @josemigallas may know more. Other than that looks nice!

Reply:
In the same file we have used refute multiple times, if need to update it to assert_not Please confirm @josemigallas  so I will update it.

Reply:
I only know what our linter tells us:
> Rails/RefuteMethods: Prefer assert_not over refute.

@nidhi-soni1104 You can run `$ bundle exec rubocop -A <file>` to run the linter and auto-fix.

Reply:
@josemigallas Will prefer assert_not over refute, so should I replace all refute present in the file, or it will work i, f I will fix the spec I have added.

Reply:
Changes done.